server/src/test/java/org/opensearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -686,6 +687,107 @@ public void testConcurrentSnapshotCreateAndDeleteOther() {
+        final String[] indexList = { "index-1", "index-2" };
+        final int shards = randomIntBetween(1, 10);
+
+        TestClusterNodes.TestClusterNode clusterManagerNode = testClusterNodes.currentClusterManager(
+            testClusterNodes.nodes.values().iterator().next().clusterService.state()
+        );
+
+        for (int i = 0; i < snapshotsList.length; i++) {
+            final StepListener<CreateSnapshotResponse> createSnapshotResponseStepListener = new StepListener<>();
+            final String snapshot = snapshotsList[i];
+            final String index = indexList[i];
+            continueOrDie(
+                createRepoAndIndex(repoName, index, shards),
+                createIndexResponse -> client().admin()
+                    .cluster()
+                    .prepareCreateSnapshot(repoName, snapshot)
+                    .setWaitForCompletion(true)
+                    .execute(createSnapshotResponseStepListener)
+            );
+
+            final StepListener<GetSnapshotsResponse> getSnapshotsResponseStepListener = new StepListener<>();
+            continueOrDie(getSnapshotsResponseStepListener, getSnapshotsResponse -> {
+                client().admin().cluster().prepareGetSnapshots(repoName).execute(getSnapshotsResponseStepListener);
+            });
+        }
+        deterministicTaskQueue.runAllRunnableTasks();
+
+        final Repository repository = clusterManagerNode.repositoriesService.repository(repoName);
+        Collection<SnapshotId> snapshotIds = getRepositoryData(repository).getSnapshotIds();
+        assertThat(snapshotIds, hasSize(2));
+        for (SnapshotId snapshotId : snapshotIds) {
+            final SnapshotInfo snapshotInfo = repository.getSnapshotInfo(snapshotId);
+            assertEquals(SnapshotState.SUCCESS, snapshotInfo.state());
+            assertEquals(0, snapshotInfo.failedShards());
+            assertTrue(Arrays.stream(snapshotsList).anyMatch(snapshotInfo.snapshotId().getName()::equals));
+        }
+
+        TransportAction getSnapshotsAction = clusterManagerNode.actions.get(GetSnapshotsAction.INSTANCE);
+        GetSnapshotsRequest repoSnapshotRequest = new GetSnapshotsRequest().repository(repoName)
+            .snapshots(snapshotsList)
+            .ignoreUnavailable(true);
+        if (getSnapshotsAction instanceof TransportGetSnapshotsAction) {
+            TransportGetSnapshotsAction transportGetSnapshotsAction = (TransportGetSnapshotsAction) getSnapshotsAction;
+            transportGetSnapshotsAction.execute(null, repoSnapshotRequest, ActionListener.wrap(repoSnapshotResponse -> {
+                assertNotNull("snapshots should be set as we are checking the current snapshot", repoSnapshotResponse.getSnapshots());
+                assertThat(repoSnapshotResponse.getSnapshots(), hasSize(2));
+            }, exception -> { throw new AssertionError(exception); }));
+        }
+    }
+
+    public void testTransportGetCurrentSnapshotOnly() {
```

Comment:
why do we need this separate test for currentSnapshot?

Reply:
CodeCoverage improvement, This has a specific code path to fall through and checks.

Reply:
can you cover current snapshot scenario above itself, you will need the code from Line 776 onwards

Reply:
not required anymore.